Sources/Workspace/PinsStore.swift

```diff
@@ -61,27 +81,45 @@ public struct PinsStore {
     public var pins: AnySequence<Pin> {
         return AnySequence<Pin>(pinsMap.values)
     }
+    
+    /// Most recent error, if any.
+    /// FIXME: This needs to be an array of diagnostics, or perhaps a structured
+    /// log of what-all went on while trying to instantiate the PinStore.
+    public var error: Error?
+    
+    /// Returns true if and only iff there are any errors; when this property is
+    /// true, mutating PinStore operations are disallowed and will throw errors.
+    /// FIXME: We should have a protocol for "things that can hold errors", and
+    /// then the `hasErrors` property should be a convenience implementation so
+    /// it doesn't need to be implemented in every adopter of the protocol.
+    public var hasError: Bool { return error != nil }
```

Comment:
I guess I am questioning whether the PinsStore is something which should "go into in an error store" versus be invalid.

In general, I think it is good to have a Result<> like thing model the "is in an error state" because it means no one can forget to check the error.

Even if PinsStore is something which can "go into an error state", I still think that might be best modeled as a PinsStore containing a Result<> wrapper around another data object. So its really just a question of whether the PinsStore should own that or clients should treat PinsStore that way, if we went that route...

For PinsStore specifically, it does feel convenient to me to allow the init to throw and treat subsequent operations which would get it into an invalid state as throwing operations which the client handles somewhat differently.

That feels very different from Workspace which I would prefer to have an error state and always be itself a valid object.

I think we should discuss in a meeting or something?